hoomd/md/pytest/test_alj.py

```diff
@@ -0,0 +1,246 @@
+        snapshot.particles.moment_inertia[:] = np.array([0, 0, moment_inertia])
+        # Not sure if this should be incircle or circumcircle;
+        # probably doesn't matter based on current usage, but may
+        # matter in the future for the potential if it's modified
+        # to actually use diameter.
+        snapshot.particles.diameter[:] = circumcircle_diameter
+    kT = 0.3
+    sim.state.thermalize_particle_momenta(hoomd.filter.All(), kT)
+
+    # Create box resize updater
+    packing_fraction = 0.4
+    final_area = hexagon.area * sim.state.N_particles / packing_fraction
+    L_final = np.sqrt(final_area)
+    final_box = hoomd.Box.square(L_final)
+
+    n_compression_start = int(1e4)
+    n_compression_end = int(1e5)
+    n_compression_total = n_compression_end - n_compression_start
+
+    box_resize = hoomd.update.BoxResize(
+        box1=sim.state.box,
+        box2=final_box,
+        trigger=int(n_compression_total / 10000),
+        variant=hoomd.variant.Ramp(0, 1, n_compression_start,
+                                   n_compression_total),
+        filter=hoomd.filter.All())
+    sim.operations += box_resize
+
+    # Define forces and methods
+    r_cut_scale = 1.3
+    kernel_scale = (1 / np.cos(np.pi / hexagon.num_vertices))
+    incircle_diameter = 2 * hexagon.incircle_radius
+    r_cut_set = incircle_diameter * kernel_scale * r_cut_scale
+
+    alj = md.pair.aniso.ALJ(default_r_cut=r_cut_set, nlist=md.nlist.Cell(0.4))
+
+    alj.shape["A"] = {
+        "vertices": hexagon.vertices,
+        "faces": [],
+        "rounding_radii": 0
+    }
+
+    alpha = 0  # Make it WCA-only (no attraction)
+    eps_att = 1.0
+    alj.params[("A", "A")] = {
+        "epsilon": eps_att,
+        "sigma_i": incircle_diameter,
+        "sigma_j": incircle_diameter,
+        "alpha": alpha
+    }
+
+    nve = md.methods.NVE(filter=hoomd.filter.All())
+    integrator = md.Integrator(dt=1e-4,
+                               forces=[alj],
+                               methods=[nve],
+                               integrate_rotational_dof=True)
+    sim.operations.integrator = integrator
+
+    # Compress box
+    sim.run(n_compression_end)
+
+    thermo = md.compute.ThermodynamicQuantities(hoomd.filter.All())
+    sim.operations += thermo
+
+    # Reset velocities after the compression, and equilibriate
+    sim.state.thermalize_particle_momenta(hoomd.filter.All(), kT)
+    sim.run(1000)
+
+    # run sim and get values back
+    storer = ConservationDataStorer(sim, thermo)
+    writer = hoomd.write.CustomWriter(action=storer, trigger=1)
+    sim.operations.writers.append(writer)
+    sim.run(1000)
+    total_energies = storer.total_energies
+    velocities = storer.velocities
+
+    # Ensure energy conservation up to the 3 digit per-particle.
+    assert np.std(total_energies) / sim.state.N_particles < 1e-3
+
+    # Test momentum conservation.
+    assert np.std(np.linalg.norm(np.sum(velocities, axis=1), axis=-1)) < 1e-6
```

Comment:
Use `max` instead of `std`. The distribution of momentum magnitudes is positive only. Therefore, any fluctuations are not about 0.

Also, for both energy and momentum you should check E-E[0] and p-p[0] instead of the actual values of E and p.

Reply:
Are E and P in this case the average values of the energy and momentum magnitude, or the last value of the energy and momentum magnitude?

Reply:
They are neither.